daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -233,6 +233,38 @@ object SBuiltin {
     }
   }
 
+  final case object SBFromTextInt64 extends SBuiltin(1) {
+    private val pattern = """[+-]?\d+""".r.pattern
+
+    def execute(args: util.ArrayList[SValue], machine: Machine): Unit = {
+      val s = args.get(0).asInstanceOf[SText].value
+      val int64 =
+        if (pattern.matcher(s).matches())
+          try {
+            Some(SInt64(java.lang.Long.parseLong(s)))
+          } catch {
+            case _: NumberFormatException =>
+              None
+          } else
+          None
+      machine.ctrl = CtrlValue(SOptional(int64))
+    }
+  }
+
+  final case object SBFromTextDecimal extends SBuiltin(1) {
+    private val pattern = """[+-]?[0-9]+(\.[0-9]+)?""".r.pattern
```

Comment:
do you allow trailing / leading 0s? this regexp does. do we know what that means when parsing?

Reply:
Please see tests 
https://github.com/digital-asset/daml/pull/1407/files#diff-ae9014deee9be354b39ebd26faa0150b

Reply:
OK, so the answer is "yes". I'm really on the fence regarding canonical representation. For users it's clearly worse... so going with the current implementation is probably the right thing.

@hurryabit what do you think?

Reply:
I'm not sure there's value in having a canonical representation for _parsing_. Being lenient seems fine to me.

Reply:
Note, we are already lenient (though a bit less than here) when parsing proto Decimal. We accept any thing that matches 
```[+-]?[0-9]{1,28}(\.[0-9]{1,10})?```